php/WP_Mock.php

```diff
@@ -546,4 +546,88 @@ public static function getDeprecatedMethodListener(): DeprecatedMethodListener
     {
         return static::$deprecatedMethodListener;
     }
+
+    /**
+     * Adds an expectation that an action should be removed.
+     *
+     * @param string $action the action hook name
+     * @param string|callable-string|callable|Type $callback the callable to be removed
+     * @param int|Type|null $priority the priority it should be registered at
+     *
+     * @return void
+     * @throws InvalidArgumentException
+     */
+    public static function expectActionRemoved(string $action, $callback, $priority = null) : void
+    {
+        self::userFunction(
+            'remove_action',
+            array(
+                'args'   => array_filter(func_get_args()),
+                'times'  => 1,
+            )
+        );
```

Comment:
For better performance, it would be ideal to be more implicit about the aguments instead of reaching for the `func_get_args`, that way the interpreter won't have to do much assumption work with the data types.

Also, we are trying to move away from the classic array syntax towards.

```suggestion
        $args = [$action, $callback];
        if ($priority) {
            $args[] = $priority;
        }

        return self::userFunction('remove_action', [
            'args'  => $args,
            'times' => 1,
        ]);
```